Refactored Convertor class and Legislation and Moved some exception …

Aasif-Gulbarga opened the pull request:
…at Their pkg place.

jurisdatum commented:
@Aasif-Gulbarga I've made a few changes. Please look them over and let me know if there's anything we should discuss

I think the RedirectException was causing confusion, so I removed it. I never really intended for it to act like a normal exception; I was just using it as a way for methods in the Legislation class to communicate with one another. But maybe that wasn't the best idea, so I reworked things so that we follow MarkLogic's redirects without throwing any exceptions.

Also, the Parameters class applies only to the legislation.xq query. If you don't want it to be an inner class, I thought I'd rename it to be LegislationParameters.

A few other little changes. I noticed that the document version field was not being set correctly in the conversion from Atom to JSON, since we switched to the MarkLogic update.

And finally, I've added some regression tests for the Atom to JSON conversion.

Aasif-Gulbarga commented:
> @Aasif-Gulbarga I've made a few changes. Please look them over and let me know if there's anything we should discuss
> 
> I think the RedirectException was causing confusion, so I removed it. I never really intended for it to act like a normal exception; I was just using it as a way for methods in the Legislation class to communicate with one another. But maybe that wasn't the best idea, so I reworked things so that we follow MarkLogic's redirects without throwing any exceptions.
> 
> Also, the Parameters class applies only to the legislation.xq query. If you don't want it to be an inner class, I thought I'd rename it to be LegislationParameters.
> 
> A few other little changes. I noticed that the document version field was not being set correctly in the conversion from Atom to JSON, since we switched to the MarkLogic update.
> 
> And finally, I've added some regression tests for the Atom to JSON conversion.

@jurisdatum Hi Jim, your changes are really adding value, I am agreeing your changes with reason. Thanks
